### PR TITLE
cpu/cc2538: enhance periph ADC

### PR DESCRIPTION
**Pull request by smlng**
Adapt the periph ADC implementation to use vendor defines where
    ever possible. Remove duplicate or obsolete defines and adapt
    board configuration as required.

### Issues/PRs references

follow up on #9691

compare #9698 and #9693

### CODE REVIEW

**Comment by smlng** [comment]
note: doesn't change binary size, its merely a cleanup.

**Comment by smlng** [comment]
added required fixes to `periph/hwrng`

File: cpu/cc2538/periph/hwrng.c [hunk 1]
```diff
@@ -21,22 +21,30 @@
...
+static cc2538_soc_adc_t *soc_adc = (cc2538_soc_adc_t *)SOC_ADC_BASE;
+
 void hwrng_init(void)
 {
     uint16_t seed = 0;
     int i;
 
     /* Make sure the RNG is on */
-    SOC_ADC->cc2538_adc_adccon1.ADCCON1bits.RCTRL = 0;
+    soc_adc->ADCCON1 = (0UL << SOC_ADC_ADCCON1_RCTRL_S);
```
Comment (A-Paul) [inline cpu/cc2538/periph/hwrng.c]: Any reason why you are shifting 0 to 0?

Comment (smlng) [reply cpu/cc2538/periph/hwrng.c]: need to ensure that RCTRL is 0, I changed this to bit unset operation.

File: cpu/cc2538/periph/hwrng.c [hunk 2]
```diff
@@ -77,11 +85,11 @@ void hwrng_read(void *buf, unsigned int num)
 
     for (count = 0; count < num; ) {
         /* Clock the RNG LSFR once: */
-        SOC_ADC->cc2538_adc_adccon1.ADCCON1bits.RCTRL = 1;
+        soc_adc->ADCCON1 = (1UL << SOC_ADC_ADCCON1_RCTRL_S);
```
[the remark this answers is not in the document]
Comment (smlng) [reply cpu/cc2538/periph/hwrng.c]: you're (kind of) right - as I remove the union with the ADCCON1bits I need to read first and then `|=` set the single bit. Good catch thanks!

**Comment by PeterKietzmann** [comment]
> The code looks fine. I can try testing later today.

@MrKevinWeiss would you?

**Comment by MrKevinWeiss** [comment]
Tested on the openmote-b board with periph_ADC.  Appears to be valid numbers.
```
2018-08-20 10:24:39,670 - INFO # ADC_LINE(0): 1014
2018-08-20 10:24:39,672 - INFO # ADC_LINE(1): 247
2018-08-20 10:24:39,686 - INFO # ADC_LINE(2): 1016
2018-08-20 10:24:39,687 - INFO # ADC_LINE(3): 1017
2018-08-20 10:24:39,688 - INFO # ADC_LINE(4): 1019
2018-08-20 10:24:39,689 - INFO # ADC_LINE(5): 68
2018-08-20 10:24:39,782 - INFO # ADC_LINE(0): 1015
2018-08-20 10:24:39,784 - INFO # ADC_LINE(1): 252
2018-08-20 10:24:39,785 - INFO # ADC_LINE(2): 1017
2018-08-20 10:24:39,786 - INFO # ADC_LINE(3): 1019
2018-08-20 10:24:39,787 - INFO # ADC_LINE(4): 1019
2018-08-20 10:24:39,787 - INFO # ADC_LINE(5): 68
```

**Comment by A-Paul** [comment]
Tested on *cc2538dk* board with *tests/periph_adc*.

3.29V:

> 2018-08-21 17:38:40,001 - INFO # Successfully initialized ADC_LINE(0)
2018-08-21 17:38:40,001 - INFO # ADC_LINE(0): 915
2018-08-21 17:38:40,096 - INFO # ADC_LINE(0): 914

0V (GND)

> 2018-08-21 17:40:22,048 - INFO # Successfully initialized ADC_LINE(0)
2018-08-21 17:40:22,048 - INFO # ADC_LINE(0): 0
2018-08-21 17:40:22,128 - INFO # ADC_LINE(0): 0

That's ~10% difference.

File: cpu/cc2538/periph/hwrng.c [hunk 3]
```diff
@@ -63,25 +72,25 @@ void hwrng_init(void)
...
-    for (count = 0; count < num; ) {
+    for (unsigned count = 0; count < num; count++) {
         /* Clock the RNG LSFR once: */
-        SOC_ADC->cc2538_adc_adccon1.ADCCON1bits.RCTRL = 1;
+        uint32_t reg32 = soc_adc->ADCCON1 | (1UL << SOC_ADC_ADCCON1_RCTRL_S);
+        soc_adc->ADCCON1 = reg32;
 
         /* Read up to 2 bytes of hwrng data: */
-        b[count++] = SOC_ADC_RNDL;
+        b[count++] = soc_adc->RNDL;
```
Comment (A-Paul) [inline cpu/cc2538/periph/hwrng.c]: You cleaned up the other inline incrementation. Putting this increment explicit on a following line would be perfect.

Comment (smlng) [reply cpu/cc2538/periph/hwrng.c]: done, I also adapted the logic of the `if` statement and to match coding style.

plus also inlined the read-write of the register as discussed offline, in a separate commit

**Comment by A-Paul** [comment]
@smlng please squash.

**Comment by smlng** [comment]
done

**Pull request merged**